docs/proposals/multiple-dns-provider-support.md

```diff
@@ -0,0 +1,82 @@
+
+## Proposed Solution
+
+MGC has three features it requires of any DNS provider in order to offer full support, DNSRecord management, Zone management and DNS Health checks.  We do not however want to limit to providers that only offer this functionality, so to add support for a provider the minimum that provider should offer is API access to managed DNS records.
+MGC will continue to provide Zone management and DNS Health checks support on a per-provider basis.
+
+Support will be added for AWS(Route53), Google(Google Cloud DNS), Azure and investigation into possible adding CoreDNS (intended for local dev purposes), with the following proposed initial support:
+
+| Provider  | DNS Records | DNS Zones | DNS Health |
+| ------------ | ------------- |  ------------- |---|
+| AWS Route53 | X | X | X |
+| Google Cloud DNS | X | X | - |
+| AzureDNS| X | X | - |
+| CoreDNS| X | - | - |
+
+
+Add DNSProvider as an API for MGC which contains all the required config for that particular provider including the credentials. This can be thought of in a similar way to a cert manager Issuer.
+Update ManagedZone to add a reference to a DNSProvider. This will be a required field on the ManagedZone and a DNSProvider must exist before a ManagedZone can be created.
+Update all controllers load the DNSProvider directly from the ManagedZone during reconciliation loops and remove the single controller wide instance. 
+Add new provider implementations for [google](assets/multiple-dns-provider-support/google/google.md), [azure](assets/multiple-dns-provider-support/azure/azure.md) and coredns.
+    * All providers constructors should accept a single struct containing all required config for that particular provider.
+    * Providers must be configured from credentials passed in the config and not rely on environment variables.
+
+## Other Solutions investigated
+
+Investigation was carried out into the suitability of [External DNS] (https://github.com/kubernetes-sigs/external-dns) as the sole means of managing dns resources.
+Unfortunately, while external dns does offer support for basic dns record management with a wide range of providers, there were too many features missing making it unsuitable at this time for integration.
+
+### External DNS as a separate controller
+
+Run external dns, as intended, as a separate controller alongside mgc, and pass all responsibility for reconciling DNSRecord resources to it. All DNSRecord reconciliation is removed from MGC.
+
+Issues:
+
+* A single instance of external dns will only work with a single provider and a single set of credentials. As it is, in order to support more than a single provider, more than one external dns instance would need to be created, one for each provider/account pair.
+* Geo and Weighted routing policies are not implemented for any provider other than AWS Route53.
+* Only supports basic dns record management (A,CNAME, NS records etc ..), with no support for managed zones of health checks.
```

Comment:
This is the crux of it really. We could run an instance per provider but then would still need to code our own solution for Geo etc